src/archive.ts

```diff
@@ -53,6 +49,12 @@ export const addZipContent = function (archive: Archiver, content: Buffer | stri
 
 // End zipping files
 export const endZip = async function (archive: Archiver, output: Writable): Promise<void> {
-  archive.finalize()
-  await pEndOfStream(output)
+  const result = new Promise<void>((resolve, reject) => {
+    output.on('error', (error) => reject(error))
+    output.on('finish', () => resolve())
+  })
+
+  await archive.finalize()
```

Comment:
We did not await this before, probably not a biggy as we anyway wait till the file is written, but better wait for both.